tests/pathPlanning/ArcPathCalculatorTest.cpp

```diff
@@ -0,0 +1,494 @@
+}
+
+TEST_F(ArcPathCalculation_DiscretizationTest, ShortTravelTest_TargetPointInsideCircle)
+{
+    const auto current_position = Point2LL{ 20'000, 10'000 };
+    const auto target_position = Point2LL{ current_position.X, current_position.Y + radius };
+    const ArcPath arc = ArcPath::calculate({ 10'000, 10'000 }, current_position, target_position, hop_height, radius, xy_speed, z_speed, step_size);
+    std::vector<std::pair<Point3LL, Velocity>> arc_points = arc.getDiscreteArc(z_height);
+    EXPECT_NEAR(vSize(Point2LL{ arc_points[0].first.x_, arc_points[0].first.y_ } - current_position), step_size, error_margin)
+        << "Fails when checking that the first point should be step_size far away from the current position.";
+    EXPECT_LT(vSize(Point2LL{ arc_points.back().first.x_, arc_points.back().first.y_ } - current_position), error_margin)
+        << "Fails when checking that the last point should be the current position.";
+    EXPECT_GE(arc_points.size(), 2) << "Fails when checking that there should be at 3 points in the discritzed arc.";
+}
+
+
+TEST_F(ArcPathCalculation_DiscretizationTest, TravelendsCloserThanStepSize)
+{
+    const coord_t low_hop_height = 1;
+    const auto current_position = Point2LL{ 10'000, 20'000 };
+    const auto target_position = Point2LL{ current_position.X + radius, current_position.Y + radius + 2 };
+    const ArcPath arc = ArcPath::calculate(
+        { 10'000, 10'000 },
+        current_position,
+        target_position,
+        low_hop_height,
+        radius,
+        xy_speed,
+        Velocity{ 1000 },
+        2 * vSize(target_position - current_position));
+    std::vector<std::pair<Point3LL, Velocity>> arc_points = arc.getDiscreteArc(z_height);
+    EXPECT_EQ(arc_points.size(), 1);
+    EXPECT_LT(vSize(Point2LL{ arc_points[0].first.x_, arc_points[0].first.y_ } - arc.end_), error_margin);
+}
+
+
+TEST_F(ArcPathCalculation_DiscretizationTest, SpeedCalculationTest_SingleArc)
+{
+    const auto current_position = Point2LL{ 20'000, 10'000 };
+    const auto target_position = Point2LL{ current_position.X + radius, current_position.Y + 2 * radius };
+    const double arc_length = (2 * std::numbers::pi * radius) / 4;
+    const coord_t total_distance = arc_length + hop_height;
+    const Velocity high_z_speed = 1.2 * hop_height / (arc_length / xy_speed);
+    const ArcPath arc = ArcPath::calculate({ 10'000, 10'000 }, current_position, target_position, hop_height, radius, xy_speed, high_z_speed, step_size);
+    const std::vector<std::pair<Point3LL, Velocity>> arc_points = arc.getDiscreteArc(z_height);
+
+    for (auto& pt : arc_points)
+    {
+        EXPECT_EQ(arc_points[0].second, pt.second) << "The speed does not stay constant over the arc.";
+    }
+    EXPECT_NEAR(arc_points[0].second * (arc_length / total_distance), xy_speed, error_margin);
+    EXPECT_LE(arc_points[0].second * (hop_height / total_distance), high_z_speed, error_margin);
```

Comment:
https://github.com/Ultimaker/CuraEngine/actions/runs/8891872064/job/24414747014?pr=2076
Hey @NadineMik can you take a look into these benchmarks and why they are failing?